packages/common/src/content/fetch.ts

```diff
@@ -174,9 +179,14 @@ export const fetchContent = async (
         options
       )
     : await fetchContentById(identifier, options);
-  // fetch record count for layers, tables, or proxied CSVs
-  const { isProxied, layer } = content;
+  // fetch record count for content that has features (e.g. layers, tables, or proxied CSVs)
+  const { layer, type } = content;
   content.recordCount =
-    isProxied || !!layer ? await fetchContentRecordCount(content) : undefined;
+    !!layer && hasFeatures(type)
+      ? await fetchContentRecordCount(content)
+      : undefined;
   return content;
 };
+
+const hasFeatures = (contentType: string) =>
```

Comment:
Style suggestion, I prefer to move these helper fns to the top of the file, right under the `import` statements, since they may one day become imports if they end up needing to move out of the file to be used elsewhere.